### PR TITLE
Related Resources refactor and External Link `scope`

### PR DESCRIPTION
**Pull request by jeffdaley**
Refactors and expands the RelatedResources component so it can be used outside of the DocumentSidebar context/format.

- Moves potentially shared types and functions (most of what happens in-modal) from `/components/documents/sidebar` → `/components`
- Adds an external-link `scope` to skip remove the "search Algolia" functionality. (There's also a document scope that replaces `allowAddingExternalLinks=false`), 
- Renames some external-link-related files and elements differentiate between `scoped` and `fallback` cases

### CODE REVIEW

File: web/app/components/document/sidebar.hbs [hunk 1]
```diff
@@ -299,7 +299,6 @@
           @documentIsDraft={{this.isDraft}}
           @productArea={{@document.product}}
           @objectID={{@document.objectID}}
-          @allowAddingExternalLinks={{true}}
```
Comment (jeffdaley) [inline web/app/components/document/sidebar.hbs:302]: This is now the default setting.

File: web/app/components/related-resources/add-or-edit-external-resource-modal.ts [hunk 2]
```diff
@@ -89,44 +100,46 @@ export default class DocumentSidebarRelatedResourcesListItemEditComponent extend
...
 
     if (this.urlIsValid) {
-      this.args.onSave(newResource);
+      const sortOrder = this.args.resource ? this.args.resource.sortOrder : 1;
+
+      this.args.onSave({
+        sortOrder,
+        name: this.title,
+        url: this.url,
+      });
```
Comment (jeffdaley) [inline web/app/components/related-resources/add-or-edit-external-resource-modal.ts:136]: Note to self: Investigate as potential cause of EmberAnimated bug

File: web/app/components/document/sidebar/related-resources/list.ts [hunk 3]
```diff
@@ -71,6 +71,17 @@ export default class DocumentSidebarRelatedResourcesListComponent extends Compon
...
+      /**
+       * For unknown reasons, Ember Animated sees an updated sprite
+       * as both inserted and removed, causing unwanted animations.
+       * This check prevents that from happening.
+       *
+       * FIXME: Investigate and fix this.
+       */
+      return;
+    }
+
```
Comment (jeffdaley) [inline web/app/components/document/sidebar/related-resources/list.ts:84]: A rather disappointing side effect of the refactor: Recently edited sprites are no longer interpreted by Ember Animated as `kept` but rather `removed` _and_ `inserted`.

I have some theories that I'll investigate soon.

File: web/app/components/document/sidebar/related-resources/list.hbs [hunk 4]
```diff
@@ -6,6 +6,8 @@
       </div>
     {{else}}
       <ul class="related-resources-list" ...attributes>
+        {{! This is being looped through twice for some reason... }}
+        {{! On immediate save, the order is wrong - maybe an array push thing? }}
```
Comment (jfreda) [inline web/app/components/document/sidebar/related-resources/list.hbs:10]: Should we follow up on this (/add a TODO)?

**Review by jfreda — APPROVED** [review]

File: web/app/components/document/sidebar/related-resources/list.hbs (hunk 4 above)
Comment (jeffdaley) [reply web/app/components/document/sidebar/related-resources/list.hbs]: This was actually resolved and I forgot to remove the comment. Thanks for catching.

**Pull request merged**